lib/libc/stdio/lib_sscanf.c

```diff
@@ -452,7 +466,9 @@ int vsscanf(FAR const char *buf, FAR const char *fmt, va_list ap)
 						width  = fwidth;
 					}
 
+#ifndef CONFIG_DYNAMIC_ALLOC_ENABLE
 					width = MIN(sizeof(tmp) - 1, width);
```

Comment:
I guess the above width limitation (128 byte) causing the string copy failure to tv buffer. Can we modify MIN check and allow long buffer copy.
We can keep tmp check and tmp buffer conversion same other than in this %s case.

Reply:
As mentioned by Sangam, the only cause of failure seems to be the statement on line 470.

width = MIN(sizeof(tmp) - 1, width);

This check is not required, since the succeeding strncpy does not copy into the variable tmp. 
If we remove this line, the issue with "%s" will be fixed. 
All other cases are already working properly and hence all other changes done under CONFIG_DYNAMIC_ALLOC_ENABLE can be reverted. 

Also there is no requirement to perform malloc in any case.
Hence, please make the above change and revert all other previous changes.

Reply:
Thanks
filtered other changes except string handling case.